jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/SelectableChannelEndPoint.java

```diff
@@ -132,7 +135,16 @@ public SocketAddress getLocalSocketAddress()
         {
             SelectableChannel channel = getChannel();
             if (channel instanceof NetworkChannel)
-                return ((NetworkChannel)channel).getLocalAddress();
+            {
+                try
+                {
+                    return ((NetworkChannel)channel).getLocalAddress();
+                }
+                catch (ClosedChannelException e)
+                {
+                    return _localSocketAddress;
+                }
+            }
```

Comment:
Don't bother with this logic here.  Just get the addresses in the constructor and set them as final fields.